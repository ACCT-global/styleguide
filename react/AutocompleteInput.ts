export { default } from './components/Autocomplete'
