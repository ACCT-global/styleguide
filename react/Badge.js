import Tag from './components/Tag/index'
import deprecated from './modules/deprecated'

export default deprecated(
  {
    useNewComponent: {
      Badge: 'Tag',
    },
  },
  Tag
)
