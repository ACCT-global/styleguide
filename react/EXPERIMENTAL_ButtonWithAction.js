import EXPERIMENTAL_ButtonWithAction from './components/EXPERIMENTAL_ButtonWithAction/index'

export default EXPERIMENTAL_ButtonWithAction
