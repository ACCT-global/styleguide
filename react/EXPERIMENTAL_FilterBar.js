import EXPERIMENTAL_FilterBar from './components/EXPERIMENTAL_FilterBar/index'

export default EXPERIMENTAL_FilterBar
