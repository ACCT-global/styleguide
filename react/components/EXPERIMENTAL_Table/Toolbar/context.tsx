import React, { FC, useContext, createContext } from 'react'

import { E2ETestable } from '../types'

const ToolbarContext = createContext<E2ETestable>(null)

export const ToolbarProvider: FC<E2ETestable> = ({ children, testId }) => {
  return (
    <ToolbarContext.Provider value={{ testId }}>
      {children}
    </ToolbarContext.Provider>
  )
}

export function useToolbarContext() {
  const context = useContext(ToolbarContext)
  if (!context) {
    throw new Error('Do not use Toolbar composites outside of context')
  }
  return context
}
