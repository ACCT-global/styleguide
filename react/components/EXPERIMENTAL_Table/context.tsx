import React, { FC, useContext, createContext } from 'react'

import { E2ETestable } from './types'
import useTableMeasures from './hooks/useTableMeasures'

type Measures = ReturnType<typeof useTableMeasures>

const TestingContext = createContext<E2ETestable>(null)
const MeasuresContext = createContext<Measures>(null)

type TableType = E2ETestable & {
  measures: Measures
}

export const TableProvider: FC<TableType> = ({
  children,
  testId,
  measures,
}) => {
  return (
    <TestingContext.Provider value={{ testId }}>
      <MeasuresContext.Provider value={measures}>
        {children}
      </MeasuresContext.Provider>
    </TestingContext.Provider>
  )
}

export function useTestingContext() {
  const context = useContext(TestingContext)
  if (!context) {
    throw new Error(
      'Do not use E2ETestable components outside of the TestingContext'
    )
  }
  return context
}

export function useMeasuresContext() {
  const context = useContext(MeasuresContext)
  if (!context) {
    throw new Error(
      'Do not use measurable components outside of the MeasuresContext'
    )
  }
  return context
}
