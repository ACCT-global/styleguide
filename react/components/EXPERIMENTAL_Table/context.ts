import { createContext } from 'react'

const Context = createContext<TableState>(null)
const { Provider } = Context

export default Context
export { Provider as TableProvider }
