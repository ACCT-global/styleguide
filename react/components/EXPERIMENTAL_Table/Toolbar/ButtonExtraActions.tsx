import React, { FC } from 'react'

import ActionMenu from '../../ActionMenu/index'
import IconOptionsDots from '../../icon/OptionsDots/index'
import { NAMESPACES, ORDER_CLASSNAMES } from '../constants'
import { MenuAction, Alignment } from './PopoverMenu'
import { ButtonVariation, ButtonSize } from './Button'
import { useButtonGroupContext } from './context'

const ButtonExtraActions: FC<ButtonExtraActionsProps> = ({
  label,
  size,
  actions,
}) => {
  const { testId } = useButtonGroupContext()
  return (
    <div
      title={label}
      className={
        ORDER_CLASSNAMES.TOOLBAR_CHILD.BUTTON_GROUP_CHILD.EXTRA_ACTIONS
      }>
      <ActionMenu
        hideCaretIcon
        buttonProps={{
          variation: ButtonVariation.Tertiary,
          id: NAMESPACES.TOOLBAR.BUTTON_EXTRA_ACTIONS,
          testId: `${testId}__extra-actions`,
          icon: (
            <span className="c-on-base">
              <IconOptionsDots />
            </span>
          ),
          size: size,
        }}
        options={actions.map(action => {
          return {
            label: action.label,
            onClick: action.onClick,
          }
        })}
      />
    </div>
  )
}

ButtonExtraActions.defaultProps = {
  size: ButtonSize.Small,
}

export type ButtonExtraActionsProps = {
  actions: Array<MenuAction>
  label?: string
  alignMenu?: Alignment
  isLoading?: boolean
  size?: ButtonSize
}

export default ButtonExtraActions
