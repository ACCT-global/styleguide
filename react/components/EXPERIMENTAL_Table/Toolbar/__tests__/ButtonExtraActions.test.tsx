import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'

import { ButtonGroupProvider } from '../context'
import ButtonExtraActions from '../ButtonExtraActions'

describe('Table V2 @ Toolbar/ButtonExtraActions spec', () => {
  const basicProps = {
    label: 'More',
    actions: [
      {
        label: 'First',
        onClick: () => {},
      },
      {
        label: 'Second',
        onClick: () => {},
      },
      {
        label: 'Third',
        onClick: () => {},
      },
    ],
  }

  function WrappedComponent() {
    return (
      <ButtonGroupProvider testId="button-group">
        <ButtonExtraActions {...basicProps} />
      </ButtonGroupProvider>
    )
  }

  it('matches snapshot', () => {
    const button = render(<WrappedComponent />)

    expect(button.asFragment()).toMatchSnapshot()
  })
  it('matches snapshot after click', () => {
    const button = render(<WrappedComponent />)
    fireEvent.click(screen.getByTestId('button-group__extra-actions'))
    expect(button.asFragment()).toMatchSnapshot()
  })
})
