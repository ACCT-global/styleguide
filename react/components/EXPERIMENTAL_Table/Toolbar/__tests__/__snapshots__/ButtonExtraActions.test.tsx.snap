// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table V2 @ Toolbar/ButtonExtraActions spec matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="order-4"
    title="More"
  >
    <div>
      <div
        class="relative"
      >
        <div>
          <button
            class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 min-h-small t-action--small icon-button dib bg-transparent b--transparent c-action-primary hover-b--transparent hover-bg-action-secondary hover-b--action-secondary pointer "
            data-testid="button-group__extra-actions"
            style="font-size: 0px;"
            tabindex="0"
            type="button"
          >
            <div
              class="vtex-button__label flex items-center justify-center h-100 ph3 "
              style="padding-top: .25em; padding-bottom: .32em;"
            >
              <span
                class="c-on-base"
              >
                <svg
                  class="vtex__icon-options-dots  "
                  fill="none"
                  height="16"
                  viewBox="0 0 32 32"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                  xmlns:xlink="http://www.w3.org/1999/xlink"
                >
                  <g
                    class="nc-icon-wrapper"
                    fill="#444444"
                  >
                    <g
                      class="nc-interact_dots-close-o-32"
                      transform="rotate(90 16 16)"
                    >
                       
                      <g
                        class="nc-dot_left"
                        transform="translate(0)"
                      >
                         
                        <circle
                          cx="6"
                          cy="16"
                          data-cap="butt"
                          data-stroke="none"
                          fill="#444444"
                          r="2"
                          stroke-linecap="butt"
                          stroke-linejoin="miter"
                        />
                         
                        <circle
                          cx="6"
                          cy="16"
                          fill="none"
                          r="2"
                          stroke="#444444"
                          stroke-linecap="square"
                          stroke-linejoin="miter"
                          stroke-width="2"
                        />
                         
                      </g>
                       
                      <g
                        class="nc-dot_right"
                        transform="translate(-0)"
                      >
                         
                        <circle
                          cx="26"
                          cy="16"
                          data-cap="butt"
                          data-stroke="none"
                          fill="#444444"
                          r="2"
                          stroke-linecap="butt"
                          stroke-linejoin="miter"
                        />
                         
                        <circle
                          cx="26"
                          cy="16"
                          fill="none"
                          r="2"
                          stroke="#444444"
                          stroke-linecap="square"
                          stroke-linejoin="miter"
                          stroke-width="2"
                        />
                         
                      </g>
                       
                      <g
                        class="nc-dot_center"
                      >
                         
                        <circle
                          cx="16"
                          cy="16"
                          data-cap="butt"
                          data-stroke="none"
                          fill="#444444"
                          r="2"
                          stroke-linecap="butt"
                          stroke-linejoin="miter"
                        />
                         
                        <circle
                          cx="16"
                          cy="16"
                          fill="none"
                          r="2"
                          stroke="#444444"
                          stroke-linecap="square"
                          stroke-linejoin="miter"
                          stroke-width="2"
                        />
                         
                      </g>
                       
                      <path
                        class="nc-line_top-right"
                        d="M16 16L27 5"
                        data-cap="none"
                        fill="none"
                        opacity="0"
                        stroke="#444444"
                        stroke-dasharray="15.56 15.56"
                        stroke-dashoffset="15.56"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                       
                      <path
                        class="nc-line_bottom-left"
                        d="M5 27l11-11"
                        data-cap="none"
                        fill="none"
                        opacity="0"
                        stroke="#444444"
                        stroke-dasharray="15.56 15.56"
                        stroke-dashoffset="-15.56"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                       
                      <path
                        class="nc-line_bottom-right"
                        d="M16 16l11 11"
                        data-cap="none"
                        fill="none"
                        opacity="0"
                        stroke="#444444"
                        stroke-dasharray="15.56 15.56"
                        stroke-dashoffset="15.56"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                       
                      <path
                        class="nc-line_top-left"
                        d="M5 5l11 11"
                        data-cap="none"
                        fill="none"
                        opacity="0"
                        stroke="#444444"
                        stroke-dasharray="15.56 15.56"
                        stroke-dashoffset="-15.56"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                       
                    </g>
                  </g>
                </svg>
              </span>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Table V2 @ Toolbar/ButtonExtraActions spec matches snapshot after click 1`] = `
<DocumentFragment>
  <div
    class="order-4"
    title="More"
  >
    <div>
      <div
        class="relative"
      >
        <div>
          <button
            class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 min-h-small t-action--small icon-button dib bg-transparent b--transparent c-action-primary hover-b--transparent hover-bg-action-secondary hover-b--action-secondary pointer "
            data-testid="button-group__extra-actions"
            style="font-size: 0px;"
            tabindex="0"
            type="button"
          >
            <div
              class="vtex-button__label flex items-center justify-center h-100 ph3 "
              style="padding-top: .25em; padding-bottom: .32em;"
            >
              <span
                class="c-on-base"
              >
                <svg
                  class="vtex__icon-options-dots  "
                  fill="none"
                  height="16"
                  viewBox="0 0 32 32"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                  xmlns:xlink="http://www.w3.org/1999/xlink"
                >
                  <g
                    class="nc-icon-wrapper"
                    fill="#444444"
                  >
                    <g
                      class="nc-interact_dots-close-o-32"
                      transform="rotate(90 16 16)"
                    >
                       
                      <g
                        class="nc-dot_left"
                        transform="translate(0)"
                      >
                         
                        <circle
                          cx="6"
                          cy="16"
                          data-cap="butt"
                          data-stroke="none"
                          fill="#444444"
                          r="2"
                          stroke-linecap="butt"
                          stroke-linejoin="miter"
                        />
                         
                        <circle
                          cx="6"
                          cy="16"
                          fill="none"
                          r="2"
                          stroke="#444444"
                          stroke-linecap="square"
                          stroke-linejoin="miter"
                          stroke-width="2"
                        />
                         
                      </g>
                       
                      <g
                        class="nc-dot_right"
                        transform="translate(-0)"
                      >
                         
                        <circle
                          cx="26"
                          cy="16"
                          data-cap="butt"
                          data-stroke="none"
                          fill="#444444"
                          r="2"
                          stroke-linecap="butt"
                          stroke-linejoin="miter"
                        />
                         
                        <circle
                          cx="26"
                          cy="16"
                          fill="none"
                          r="2"
                          stroke="#444444"
                          stroke-linecap="square"
                          stroke-linejoin="miter"
                          stroke-width="2"
                        />
                         
                      </g>
                       
                      <g
                        class="nc-dot_center"
                      >
                         
                        <circle
                          cx="16"
                          cy="16"
                          data-cap="butt"
                          data-stroke="none"
                          fill="#444444"
                          r="2"
                          stroke-linecap="butt"
                          stroke-linejoin="miter"
                        />
                         
                        <circle
                          cx="16"
                          cy="16"
                          fill="none"
                          r="2"
                          stroke="#444444"
                          stroke-linecap="square"
                          stroke-linejoin="miter"
                          stroke-width="2"
                        />
                         
                      </g>
                       
                      <path
                        class="nc-line_top-right"
                        d="M16 16L27 5"
                        data-cap="none"
                        fill="none"
                        opacity="0"
                        stroke="#444444"
                        stroke-dasharray="15.56 15.56"
                        stroke-dashoffset="15.56"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                       
                      <path
                        class="nc-line_bottom-left"
                        d="M5 27l11-11"
                        data-cap="none"
                        fill="none"
                        opacity="0"
                        stroke="#444444"
                        stroke-dasharray="15.56 15.56"
                        stroke-dashoffset="-15.56"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                       
                      <path
                        class="nc-line_bottom-right"
                        d="M16 16l11 11"
                        data-cap="none"
                        fill="none"
                        opacity="0"
                        stroke="#444444"
                        stroke-dasharray="15.56 15.56"
                        stroke-dashoffset="15.56"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                       
                      <path
                        class="nc-line_top-left"
                        d="M5 5l11 11"
                        data-cap="none"
                        fill="none"
                        opacity="0"
                        stroke="#444444"
                        stroke-dasharray="15.56 15.56"
                        stroke-dashoffset="-15.56"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                       
                    </g>
                  </g>
                </svg>
              </span>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
